spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/LSTM2.scala

```diff
@@ -0,0 +1,191 @@
+      wRegularizer = wRegularizer, bRegularizer = bRegularizer).inputs(input2)
+
+    val linearh2g4 = Linear(hiddenSize, hiddenSize,
+      wRegularizer = wRegularizer, bRegularizer = bRegularizer).inputs(input2)
+
+    h2g = JoinTable(1, 1).inputs(linearh2g1, linearh2g2, linearh2g3, linearh2g4)
+
+    val caddTable = CAddTable(false).inputs(i2g, h2g)
+    val reshape = Reshape(Array(4, hiddenSize)).inputs(caddTable)
+    val split1 = Select(2, 1).inputs(reshape)
+    val split2 = Select(2, 2).inputs(reshape)
+    val split3 = Select(2, 3).inputs(reshape)
+    val split4 = Select(2, 4).inputs(reshape)
+
+    // make different instances of inner activation
+    val innerActivation2 = innerActivation.cloneModule()
+    innerActivation2.setName(Integer.toHexString(java.util.UUID.randomUUID().hashCode()))
+    val innerActivation3 = innerActivation.cloneModule()
+    innerActivation3.setName(Integer.toHexString(java.util.UUID.randomUUID().hashCode()))
+
+    (innerActivation.inputs(split1),
+      activation.inputs(split2),
+      innerActivation2.inputs(split3),
+      innerActivation3.inputs(split4))
+  }
+
+  def buildModel(): Sequential[T] = {
+    Sequential()
+      .add(FlattenTable())
+      .add(buildLSTM2())
+      .add(ConcatTable()
+        .add(SelectTable(1))
+        .add(NarrowTable(2, 2)))
+  }
+
+  def buildLSTM2(): Graph[T] = {
+    val input1 = Input()
+    val input2 = Input()
+    val input3 = Input()
+    val (in, hid, forg, out) = buildGates()(input1, input2)
+
+    /**
+     * g: activation
+     * cMult1 = in * hid
+     * cMult2 = forg * input3
+     * cMult3 = out * g(cMult1 + cMult2)
+     */
+    val cMult1 = CMulTable().inputs(in, hid)
+    val cMult2 = CMulTable().inputs(forg, input3)
+    val cadd = CAddTable(true).inputs(cMult1, cMult2)
+    val activation2 = activation.cloneModule()
+    activation2.setName(Integer.toHexString(java.util.UUID.randomUUID().hashCode()))
+    val activate = activation2.inputs(cadd)
+    val cMult3 = CMulTable().inputs(activate, out)
+
+    val out1 = Identity().inputs(cMult3)
+    val out2 = Identity().inputs(cMult3)
+    val out3 = cadd
+
+    /**
+     * out1 = cMult3
+     * out2 = out1
+     * out3 = cMult1 + cMult2
+     */
+    Graph(Array(input1, input2, input3), Array(out1, out2, out3))
+  }
+
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[LSTM2[T]]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: LSTM2[T] =>
+      super.equals(that) &&
+        (that canEqual this) &&
+        inputSize == that.inputSize &&
+        hiddenSize == that.hiddenSize
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(super.hashCode(), inputSize, hiddenSize)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+  }
+
+  override def reset(): Unit = {
+    super.reset()
+    cell.reset()
+  }
+
+
+  override def toString: String = s"LSTM2($inputSize, $hiddenSize)"
+}
+
+object LSTM2 {
```

Comment:
This layer is just for unit tests, right?
If so, you should move it to test package.